### PR TITLE
add tests for using new menuLayout parameter

### PR DESCRIPTION
**Pull request by iCollin**
ATF Test Scripts to check [#2925](https://github.com/smartdevicelink/sdl_core/issues/2925)

This PR is **ready** for review.

### Summary
Add the ability to request a tiled menu through SetGlobalProperties or AddSubMenu RPCs.

### ATF version
develop

##### Enhancements
* Adds new parameter menuLayout to SetGlobalProperties RPC.
* Adds new parameter menuLayout to AddSubMenu RPC.
* Adds new parameter menuLayoutsAvailable to HMI Capabilities

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Smoke/API/050_SetMenuLayout_unspported_WARNINGS.lua [hunk 1]
```diff
@@ -0,0 +1,79 @@
+---------------------------------------------------------------------------------------------------
```
Comment (JackLivio) [inline test_scripts/Smoke/API/050_SetMenuLayout_unspported_WARNINGS.lua]: typo in the name "unspported"

**Comment by JackLivio** [comment]
Please put request and response parameter objects under the section

> --[[ Local Variables ]]

And move the functions to a section names 

> --[[ Local Functions ]]

[Example test script format](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2196/files#diff-4d4a4526406b63d637131ea307796dd9)

File: test_scripts/Smoke/API/050_SetMenuLayout_unspported_WARNINGS.lua (hunk 1 above)
Comment (iCollin) [reply test_scripts/Smoke/API/050_SetMenuLayout_unspported_WARNINGS.lua]: fixed in https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2213/commits/9ff4d86aca9eae938af767e253c57d09fa0fec37

**Comment by theresalech** [comment]
Please ensure implementation reflects the revisions included in accepted Evolution Proposal [SDL 0231 Revisions - Add Tiles as an Option for Main Menus](https://github.com/smartdevicelink/sdl_evolution/issues/800).  This is a revision to remove the default value (`defvalue="LIST"`) from the menuLayout parameter introduced by the original proposal.  Please reference issue and proposal markdown file for more details.

**Pull request merged**